### PR TITLE
MGMT-9334: Add an endpoint to download an ipxe-boot script

### PR DESCRIPTION
**Pull request by carbonin**
As described in [the enhancement](https://github.com/openshift/assisted-service/blob/c5eacda676475f5a6de123678c1af353a2368bd3/docs/enhancements/ipxe-host-boot.md
) this PR implements the ipxe-boot script download endpoint.

Specifically `GET /v2/infra-envs/{infra-env-id}/downloads/files?file_name=ipxe-script`.

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

Deployed assisted service locally with podman, created an infra-env, called the new endpoint.

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @filanov 
/cc @CrystalChun 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

### CODE REVIEW

**Comment by filanov** [comment]
@avishayt please take a look at the swagger.

File: swagger.yaml [hunk 1]
```diff
@@ -3509,6 +3509,62 @@ paths:
...
+      tags:
+        - installer
+      security:
+        - userAuth: [admin, read-only-admin, user]
+        - urlAuth: []
+        - imageAuth: []
+      description: Creates an iPXE boot script for the infra-env.
+      operationId: GetInfraEnvIPXEScript
+      produces:
+        - 'application/octet-stream'
```
Comment (avishayt) [inline swagger.yaml:3523]: :+1: :-)

File: swagger.yaml [hunk 2]
```diff
@@ -3509,6 +3509,62 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+  /v2/infra-envs/{infra_env_id}/downloads/ipxe-script:
```
Comment (avishayt) [inline swagger.yaml:3512]: Any reason not to put it under /v2/infra-envs/{infra_env_id}/downloads/files ?

Comment (carbonin) [reply swagger.yaml:3512]: Not really, it just didn't come to mind. Want me to change it? (I'll also update the enhancement)

**Review by filanov — APPROVED** [review]

**Comment by filanov** [comment]
/override ci/prow/e2e-metal-assisted-operator-ztp
The test is currently broken.

**Pull request merged**